src/prefect/server/api/validation.py

```diff
@@ -0,0 +1,47 @@
+from prefect._vendor.fastapi import HTTPException, status
+
+import prefect.server.schemas as schemas
+from prefect.utilities.validation import validate_values_conform_to_schema
+
+
+def _get_base_config_defaults(base_config: dict):
+    template: dict = base_config.get("variables", {}).get("properties", {})
+    defaults = dict()
+    for variable_name, attrs in template.items():
+        if "default" in attrs:
+            defaults[variable_name] = attrs["default"]
+
+    return defaults
+
+
+def _merge_job_variables(base_vars: dict, deployment_vars: dict, flow_run_vars: dict):
+    result = {}
+    result.update(base_vars)
+    result.update(deployment_vars)
+    result.update(flow_run_vars)
+    return result
```

Comment:
super nit: is this fn necessary? seems like this is just
```python
{**base_vars, **deployment_vars, **flow_run_vars}
```

Reply:
I think you're right about that! I went the `update` route to make the order explicit but the unpacking syntax is clear in that case too.